mls-rs/src/client_builder.rs

```diff
@@ -297,56 +297,6 @@ impl<C: IntoConfig> ClientBuilder<C> {
         ClientBuilder(c)
     }
 
-    /// Add a key package extension to the list of key package extensions supported by the client.
-    pub fn key_package_extension<T>(
-        self,
-        extension: T,
-    ) -> Result<ClientBuilder<IntoConfigOutput<C>>, ExtensionError>
-    where
-        T: MlsExtension,
-        Self: Sized,
-    {
-        let mut c = self.0.into_config();
-        c.0.settings.key_package_extensions.set_from(extension)?;
-        Ok(ClientBuilder(c))
-    }
-
-    /// Add multiple key package extensions to the list of key package extensions supported by the
-    /// client.
-    pub fn key_package_extensions(
-        self,
-        extensions: ExtensionList,
-    ) -> ClientBuilder<IntoConfigOutput<C>> {
-        let mut c = self.0.into_config();
-        c.0.settings.key_package_extensions.append(extensions);
-        ClientBuilder(c)
-    }
-
-    /// Add a leaf node extension to the list of leaf node extensions supported by the client.
-    pub fn leaf_node_extension<T>(
-        self,
-        extension: T,
-    ) -> Result<ClientBuilder<IntoConfigOutput<C>>, ExtensionError>
-    where
-        T: MlsExtension,
-        Self: Sized,
-    {
-        let mut c = self.0.into_config();
-        c.0.settings.leaf_node_extensions.set_from(extension)?;
-        Ok(ClientBuilder(c))
-    }
-
-    /// Add multiple leaf node extensions to the list of leaf node extensions supported by the
-    /// client.
-    pub fn leaf_node_extensions(
-        self,
-        extensions: ExtensionList,
-    ) -> ClientBuilder<IntoConfigOutput<C>> {
-        let mut c = self.0.into_config();
-        c.0.settings.leaf_node_extensions.append(extensions);
-        ClientBuilder(c)
-    }
-
     /// Set the lifetime duration in seconds of key packages generated by the client.
     pub fn key_package_lifetime(self, duration_in_s: u64) -> ClientBuilder<IntoConfigOutput<C>> {
         let mut c = self.0.into_config();
```

Comment:
I wonder if this should be flexible too? E.g. longer lifetime for last resort key package?

Reply:
Makes sense. I'll extract it.

Reply:
So, after trying to extract this I ran into quite a few unforeseen issues. Despite this parameter being named `key_package_lifetime`, It seems to be used whenever we create a node in the tree, including places like group creation where a KP is not normally involved in the formal sense. This made extraction quite cumbersome.

I'm open to discussing extracting this but would push for that to be a separate issue given the above. Thoughts?

Reply:
If it's used to create leaf nodes for updates, that's very strange. If you could create an issue to investigate, that would be great. Definitely not part of this PR.

Reply:
Filed https://github.com/awslabs/mls-rs/issues/197

Reply:
Thank you!